lib/validatex/validators.ex

```diff
@@ -71,4 +243,79 @@ defmodule Validatex.Validators do
       {:error, msg}
     end
   end
+
+  # Private
+
+  defp validate(value, req_val, msg, fun) when is_function(fun, 3) do
```

Comment:
toto je moc obecna fce, ktera se snazi resit jak integer, tak float dohromady, pritom by stacilo, aby fce kteree volaji tuto fci nemeli guard `is_number`, ale aby se rozdelili na dalsi dve hlavy a to s guardem `is_integer`, resp. `is_float` a do teto fce se pak predala spravna _cast_ fce `integer`, resp. `float`.

Reply:
a nestacilo by spis mit pouze dve fce `validate` jednu pro `integer`, druhou pro `float`?

Reply:
Hmm tak tak on je problem ten, ze to `value` je `string`  a me zatim nic jineho co by vypadalo lepsejc nenapadlo, nez jak je to ted :-(